lib/Sema/CodeSynthesis.cpp

```diff
@@ -2463,8 +2461,8 @@ synthesizeSetterBody(AccessorDecl *setter, ASTContext &ctx) {
     if (var->getAttrs().hasAttribute<LazyAttr>()) {
       // Lazy property setters write to the underlying storage.
       auto *storage = var->getLazyStorageProperty();
-      return synthesizeTrivialSetterBodyWithStorage(setter, TargetImpl::Storage,
-                                                    storage, ctx);
+      return synthesizeObservedSetterBody(setter, TargetImpl::Storage, ctx,
```

Comment:
This should still call synthesizeTrivialSetterBodyWithStorage() if there are no observers, I think, unless synthesizeObservedSetterBody() just degrades to an ordinary setter in that case.

Reply:
I think it's the same - the observers are only added if there is a willSet/didSet function.